packages/ra-core/src/sideEffect/undo.js

```diff
@@ -34,7 +34,10 @@ export function* handleUndoRace(undoableAction) {
         // if not cancelled, redispatch the action, this time immediate, and without success side effect
         yield put({
             ...action,
-            meta: metaWithoutSuccessSideEffects,
+            meta: {
+                ...metaWithoutSuccessSideEffects,
+                onSuccess: { refresh: true },
```

Comment:
The forced refresh allows to make sure the displayed data is up to date with the server data after an optimistic operation.